mypy/messages.py

```diff
@@ -182,28 +182,39 @@ def is_errors(self) -> bool:
 
     def report(self, msg: str, context: Optional[Context], severity: str,
                file: Optional[str] = None, origin: Optional[Context] = None,
-               offset: int = 0) -> None:
+               offset: int = 0, strip_msg: bool = True) -> None:
         """Report an error or note (unless disabled)."""
         if self.disable_count <= 0:
+            msg = msg.strip() if strip_msg else msg
             self.errors.report(context.get_line() if context else -1,
                                context.get_column() if context else -1,
-                               msg.strip(), severity=severity, file=file, offset=offset,
```

Comment:
After some thinking I am not sure why do we actually need to strip the message here. How many tests fail if you remove the `.strip()` here? Maybe we can instead update our test framework?